static/app/components/dropdownMenuItemV2.tsx

```diff
@@ -105,137 +111,160 @@ type Props = {
   renderAs?: React.ElementType;
 };
 
+/**
+ * Wraps withRouter and forwardRef around Component
+ */
+const withRouterForwardRef = Component => {
+  const WithRouter = withRouter(
+    ({
+      forwardedRef,
+      ...props
+    }: Props &
+      WithRouterProps & {
+        forwardedRef: React.ForwardedRef<React.RefObject<HTMLLIElement>>;
+      }) => <Component ref={forwardedRef} {...props} />
+  );
+
+  return forwardRef<React.RefObject<HTMLLIElement>, Props>((props, ref) => (
+    <WithRouter {...props} forwardedRef={ref} />
+  ));
+};
+
```

Comment:
Using `withRouter` to handle internal navigation. Sadly, `withRouter` doesn't forward refs automatically, so we have to use this little function.

Reply:
should we move this helper function to `app/utils`?

Reply:
hmm afaik we only need both `withRouter` and `forwardRef` in this component, so not sure how useful this would be in `utils`. 

also, do we intend to upgrade to `react-router` v6 soon? Cause in newer versions we can just use the `useNavigate()` hook instead of this HOC.

Reply:
I understand, but it may happen that in the future someone needs to use the same wrapper and if it's not in 
`app/utils` maybe he doesn't know that this wrapper already exists. 

About updating react-router I can say that this is a dream 🤩  we were investigating but this has been a while and I don't know what the current status is... maybe @evanpurkhiser has more info on this